### PR TITLE
Java Security Manager and scripting

### PR DESCRIPTION
**Pull request by clintongormley**
Added docs explaining the impact of the Java Security Manager on scripting
languages, how to disable the JSM, and how to customise the classloader
whitelist.

Closes https://github.com/elastic/elasticsearch/issues/16094
Closes https://github.com/elastic/elasticsearch/issues/14290

### CODE REVIEW

**Comment by clintongormley** [comment]
@rmuir please could you review?

File: docs/reference/modules/scripting/security.asciidoc [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+In a script, attempting to load a class that does not appear in the whitelist
+_may_ result in a `ClassNotFoundException`, for instance this script:
+
+[source,json]
+------------------------------
+GET _search
+{
+  "script_fields": {
+    "the_hour": {
+      "script": "use(java.math.BigInteger); new BigInteger(1)"
```
Comment (clintongormley) [inline docs/reference/modules/scripting/security.asciidoc]: I use BigInteger here as it is not a whiltelisted class, but I see that BigDecimal is.  Is this a mistake?

Comment (rmuir) [inline docs/reference/modules/scripting/security.asciidoc]: the list from the old "groovy sandbox" was used mostly as-is (with a few missing classes added like `java.lang.Object` and `java.lang.String`. I think BigDecimal was in the list, as it is implicitly used as the result of integer division.

**Comment by rmuir** [comment]
+1, this looks great.

**Comment by clintongormley** [comment]
thanks @rmuir

**Pull request merged**